src/resmom/linux/alps.c

```diff
@@ -7011,18 +7012,21 @@ create_vnodes_KNL(basil_response_query_system_t *sys_knl)
  *	  they pertain to KNL Nodes.
  *
  * @param[in] ptrNodeGrp Pointer to the current Node Group in the System XML Response.
+ * @param[in] skip_state_check Indicates wheather this function to look for the KNL node state.
  *
  * @return int
  * @retval 1 Indicates that the Node Group should not be considered.
  * @retval 0 Indicates that the Node Group should be considered.
  *
  */
 static int
-exclude_from_KNL_processing(basil_system_element_t *ptrNodeGrp)
+exclude_from_KNL_processing(basil_system_element_t *ptrNodeGrp,
+		short int skip_state_check)
 {
 	if ((strcmp(ptrNodeGrp->role, BASIL_VAL_BATCH_SYS) != 0) ||
-	    (strcmp(ptrNodeGrp->state, BASIL_VAL_UP_SYS) != 0) ||
-	    ((strcmp(ptrNodeGrp->numa_cfg, "") == 0) &&
+	    (skip_state_check ? (strcmp(ptrNodeGrp->state,
```

Comment:
the variable is doing exactly the opposite of what the name suggests. could you please change it to check_state.

Reply:
Done